openlibrary/core/vendors.py

```diff
@@ -272,10 +272,13 @@ def _get_amazon_metadata(id_, id_type='isbn', resources=None):
 
     try:
         r = requests.get('http://%s/isbn/%s' % (affiliate_server_url, id_))
+        r.raise_for_status()
         return r.json().get('hit') or None
     except requests.exceptions.ConnectionError:
         logger.exception("Affiliate Server unreachable")
-        return None
+    except requests.exceptions.HTTPError:
+        logger.exception("Affiliate Server: id {} not found".format(id_))
```

Comment:
This message will be misleading for error response codes other than `404`.  It may be better to include the response code and the reason in the log message.

Reply:
https://docs.python.org/3/library/logging.html#logging.exception writes a full stack trace (https://docs.python.org/3/library/traceback.html) to the log so that all exception details are available.